app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -186,172 +183,254 @@ private fun MessageComposer(
-                        transition,
-                        messageComposerState,
-                        focusManager,
-                        membersToMention.isNotEmpty()
+                        .fillMaxWidth()
+                        .wrapContentHeight()
+                        .padding(bottom = dimensions().spacing8x)
+                        .weight(1f)
+                ) {
+                    messagesContent()
+                    MembersMentionList(
+                        membersToMention = membersToMention,
+                        onMentionPicked = onMentionPicked
                     )
                 }
+
+                MessageComposerInput(
+                    transition = transition,
+                    messageComposerInnerState = messageComposerState,
+                    interactionAvailability = interactionAvailability,
+                    securityClassificationType = securityClassificationType,
+                    membersToMention = membersToMention,
+                    onMentionPicked = onMentionPicked,
+                    onSendButtonClicked = onSendButtonClicked,
+                    onToggleFullScreen = messageComposerState::toggleFullScreen,
+                    onCancelReply = messageComposerState::cancelReply,
+                )
             }
+
             // Box wrapping for additional options content
             // we want to offset the AttachmentOptionsComponent equal to where
             // the device keyboard is displayed, so that when the keyboard is closed,
             // we get the effect of overlapping it
             if (messageComposerState.attachmentOptionsDisplayed && interactionAvailability == InteractionAvailability.ENABLED) {
                 AttachmentOptions(
-                    keyboardHeight = keyboardHeight,
-                    messageComposerState = messageComposerState,
+                    attachmentInnerState = messageComposerState.attachmentInnerState,
                     onSendAttachment = onSendAttachmentClicked,
                     onMessageComposerError = onMessageComposerError,
                     isFileSharingEnabled = isFileSharingEnabled,
-                    tempCachePath = tempCachePath
+                    tempCachePath = tempCachePath,
+                    Modifier
+                        .constrainAs(attachmentOptions) {
+                            top.linkTo(topOfKeyboardGuideLine)
+                            bottom.linkTo(parent.bottom)
+                            start.linkTo(parent.start)
+                            end.linkTo(parent.end)
+
+                            height = Dimension.fillToConstraints
+                        }
+                        .fillMaxWidth()
+
+                )
+            }
+        }
+
+    }
+
+    BackHandler(messageComposerState.attachmentOptionsDisplayed) {
+        messageComposerState.hideAttachmentOptions()
+        messageComposerState.toEnabled()
+        messageComposerState.cancelReply()
```

Comment:
the same here: why do we cancel reply on backPress when attachments are visible?

Reply:
good question 🤔

Reply:
it's removed now, it shouldn't cancel the replyit's removed, it shouldn't cancel the reply